pkg/config/types/downloaders.go

```diff
@@ -0,0 +1,31 @@
+package types
+
+import (
+	"slices"
+	"strings"
+)
+
+var _ Provider = (*ResultDownloaders)(nil)
+
+type ResultDownloaders struct {
+	Disabled []string       `yaml:"Disabled,omitempty"`
+	Timeout  Duration       `yaml:"Timeout,omitempty"`
+	IPFS     IpfsDownloader `yaml:"IPFS,omitempty"`
+}
+
+func (r ResultDownloaders) Enabled(kind string) bool {
+	return !slices.ContainsFunc(r.Disabled, func(s string) bool {
+		return strings.ToLower(s) == strings.ToLower(kind)
+	})
+}
+
+var _ Configurable = (*IpfsDownloader)(nil)
+
+type IpfsDownloader struct {
+	// Endpoint is the multiaddress to connect to for IPFS.
+	Endpoint string `yaml:"Endpoint,omitempty"`
+}
+
+func (i IpfsDownloader) Installed() bool {
+	return i != IpfsDownloader{}
+}
```

Comment:
> is this the right place to implemenet this?

Yes, I believe so unless there is an alternative you have in mind? 

Looking at the internals of the `IPFSDownloader`:
```go
type Downloader struct {
	client *ipfs.Client
}

func NewIPFSDownloader(client *ipfs.Client) *Downloader {
	return &Downloader{
		client: client,
	}
}

func (d *Downloader) IsInstalled(context.Context) (bool, error) {
	return d.client != nil, nil
}

```
Along with how we instantiate it:
```go
	if cfg.Enabled(models.StorageSourceIPFS) {
		if cfg.IPFS.Installed() {
			ipfsClient, err := ipfs_client.NewClient(ctx, cfg.IPFS.Endpoint)
			if err != nil {
				return nil, err
			}
			providers[models.StorageSourceIPFS] = ipfs.NewIPFSDownloader(ipfsClient)
		}
	}

```

Would you prefer we pass a nil client to `NewIPFSDownloader`?

Reply:
I see what you mean. IPFS is a special case in our case base as it accepts a client instead of config. We should change things to make it accept a config instead. For now, lets have the same previous behaviour as in 
```
if ipfsConnect != "" {
		ipfsClient, err := ipfs_client.NewClient(ctx, ipfsConnect)
		if err != nil {
			return nil, err
		}
		providers[models.StorageSourceIPFS] = ipfs.NewIPFSDownloader(ipfsClient)
	}
```

No need to introduce `IsConfigured` as it doens't apply to other engines